vignettes/advanced.Rmd

```diff
@@ -26,102 +29,290 @@ knitr::opts_chunk$set(
+
+```
+
+Let's assume that the chosen multiple imputation (MI) method is the approximate Bayesian MI with 10 imputations that can be set with the function `method_approxbayes()` as follows:
+
+```{r}
+method <- method_approxbayes(n_sample = 10)
+```
+
+We can now call sequentially the 4 key functions of `rbmi` to perform the multiple imputation.
+Please note that the management of post-discontinuation observed data is performed without additional complexity for the user.
+`draws()` automatically excludes post-ICE data handled with a reference-based method using information provided by the argument `data_ice`.
+`impute()` will impute only actual missing data in `data[[vars$outcome]]`.
+
+```{r}
+set.seed(123)
+draw_obj <- draws(
+    data = data,
+    data_ice = data_ice_CIR,
+    vars = vars,
+    method = method
+)
+
+impute_obj_CIR <- impute(
+    draw_obj,
+    references = c("Control" = "Control", "Intervention" = "Intervention")
 )
+
+ana_obj_CIR <- analyse(
+    impute_obj_CIR,
+    vars = vars_an
+)
+
+pool(ana_obj_CIR)
 ```
 
+# Adding time-varying covariates in the imputation model
+
+`rbmi` allows to include time-varying covariates in the imputation model.
+For example, lets suppose that we want to implement an estimator equivalent to the one described in section \@ref(sec:postICEobs) with the following differences:
+
+- There is an additional time-varying covariate that represents the number of visits from the ICE treatment discontinuation. This is set equal to 0 up to the treatment discontinuation and to the time from treatment discontinuation
+at subsequent visits. Additionally a treatment by time-varying covariate interaction terms is included in the model. This variable allows to estimate the post-ICE trajectory in each arm directly from the data, without explicitly making apriori any (reference-based) assumption.
+- The imputation is performed under MAR.
+
+Indeed, one could assume that post-ICE data are missing-at-random after taking into account baseline characteristics, observed outcomes and additional information regarding the time of the ICE. For a precise description of this approach, we refer to @Guizzaro2021.
+
+Let's first compute the time-varying covariate:
+
+```{r}
+data <- data %>%
+  group_by(id) %>%
+  mutate(time_from_ice1 = cumsum(ind_ice1))
+```
+
+and include it in the imputation model, crossed with the `group` variable:
+
+```{r}
+vars_tv <- set_vars(
+    subjid = "id",
+    visit = "visit",
+    outcome = "change",
+    group = "group",
+    covariates = c("visit*outcome_bl", "visit*group", "time_from_ice1*group"),
+    strategy = "strategy"
+    )
+```
+
+We can now call sequentially the 4 key `rbmi` functions:
+
+```{r}
+draw_obj <- draws(
+    data = data,
+    data_ice = NULL, # if NULL, MAR is assumed for every missing data
+    vars = vars_tv,
+    method = method
+)
+
+impute_obj_tv <- impute(
+    draw_obj,
+    references = c("Control" = "Control", "Intervention" = "Intervention")
```

Comment:
Is this correct ? As in are both groups meant to reference themselves ?

Reply:
Yes, I think this is correct, since we impute under MAR.

Reply:
We might change this and allow user to not need to specify `references` when imputation is under MAR